game/final_obs_gen.py

```diff
@@ -0,0 +1,494 @@
+##################################################
+# Imports
+##################################################
+
+import sys
+import random
+import numpy as np
+import pygame
+
+
+##################################################
+# Class Modules
+##################################################
+
+
+class Game:
+
+    def __init__(self):
+
+        # Display Constants
+        self.GRID_SIZE = 6  # scale entire pygame panel size
+        self.U_WIDTH, self.U_HEIGHT = 150, 55  # full panel proportions (20x19 dino)
+        self.REF, self.LOW_BIRD, self.HIGH_BIRD = 2, 18, 28  # proportions for the axis
+        self.BG_COLOR = (255, 255, 255)  # white colour (background)
+        self.BLOCK_COLOR = (0, 0, 0)  # black colour (sprites)
+
+        # Timing Constants
+        self.FPS = 60  # frames per second
+        self.UPDATE_WALK_FRAMES = 6  # how many frames to wait before dino switches legs
+
+        # Setup Axis
+        self.WIDTH, self.HEIGHT = (
+            self.U_WIDTH * self.GRID_SIZE,
+            self.U_HEIGHT * self.GRID_SIZE,
+        )
+        self.REF_AXIS = self.HEIGHT - self.REF * self.GRID_SIZE
+        self.LOW_BIRD_AXIS = self.HEIGHT - self.LOW_BIRD * self.GRID_SIZE
+        self.HIGH_BIRD_AXIS = self.HEIGHT - self.HIGH_BIRD * self.GRID_SIZE

```

Comment:
We changed the implementations of the screen now! There are still axis variables to write using the bottom left corner as 0 0, as opposed to 0, 0 being the top left corner so that it's easier to understand.